scripts/openstack-update.sh

```diff
@@ -0,0 +1,122 @@
+    OPENSTACK_DEPLOYED_VERSION=$(oc get -n $NAMESPACE $OPENSTACK_VERSION_CR --template={{.spec.targetVersion}})
+    sed -i $OUTFILE -e "s|value: $OPENSTACK_DEPLOYED_VERSION|value: $VERSION|"
+
+    oc apply -f $OUTFILE
+
+    # wait for openstack-operator to start reinstallation
+    until $(oc get csv -l operators.coreos.com/openstack-operator.openstack-operators -n openstack-operators | grep -q Installing); do
+        sleep 1;
+    done
+
+    # wait for openstack-operator to installation to complete
+    until $(oc get csv -l operators.coreos.com/openstack-operator.openstack-operators -n openstack-operators | grep -q Succeeded); do
+        sleep 1;
+    done
+fi
+
+# wait until openstackversion cr completes reconcile, status.availableVersion should be the same as VERSION
+until $(oc get -n openstack $OPENSTACK_VERSION_CR  --template={{.status.availableVersion}} | grep -q $VERSION); do
+    sleep 1;
+done
+
+oc project $NAMESPACE
+
+OPENSTACK_DEPLOYED_VERSION=$(oc get $OPENSTACK_VERSION_CR --template={{.spec.targetVersion}})
+
+cat <<EOF >openstackversionpatch.yaml
+    "spec": {
+      "targetVersion": "$VERSION"
+      }
+EOF
+
+oc patch $OPENSTACK_VERSION_CR  --type=merge  --patch-file openstackversionpatch.yaml
+
+# wait for ovn update on control plane
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateOVNControlplane --timeout=$TIMEOUT
+
+# start ovn update on data plane
+DATAPLANE_NODESET=$(oc get openstackdataplanenodeset -o name | awk -F'/' '{print $2}')
+DATAPLANE_DEPLOYMENT=$(oc get openstackdataplanedeployment -o name | awk -F'/' '{print $2}')
+
+cat <<EOF >edpm-deployment-ovn-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-ovn-update
+spec:
+  nodeSets:
+    - $DATAPLANE_NODESET
+  servicesOverride:
+    - ovn
+EOF
+
+oc create -f edpm-deployment-ovn-update.yaml
+
+oc get openstackdataplanedeployment
+# wait for ovn dataplane completes
+oc wait $OPENSTACK_VERSION_CR  --for=condition=MinorUpdateOVNDataplane --timeout=$TIMEOUT
+echo "MinorUpdateOVNDataplane completed"
+
+# wait for control plane update to complete
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateControlplane --timeout=$TIMEOUT
+
+# TODO: fix docs to include step
+# create openstackdataplaneservice update
+
+oc apply -f https://raw.githubusercontent.com/openstack-k8s-operators/openstack-operator/main/config/services/dataplane_v1beta1_openstackdataplaneservice_update.yaml
+
+# start data plane plane update for rest of edpm services
+cat <<EOF >edpm-deployment-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-update
+spec:
+  nodeSets:
+    - $DATAPLANE_NODESET
+  servicesOverride:
+    - update
+EOF
+
+oc create -f edpm-deployment-update.yaml
+
+# wait for completion of minor update
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateDataplane --timeout=$TIMEOUT
```

Comment:
So after this, the dataplane's rpms are updated, but not the container images (beside the ovn-controller). Am I missing something 

```
ceilometer_agent_compute quay.io/podified-antelope-centos9/openstack-ceilometer-compute@sha256:be31ef6df0eb54fb6a1f71d5f746b61312c0ea6782c1e8ffca12cebf283478a4 2024-06-27 09:15:16.238682865 +0000 UTC
ceilometer_agent_ipmi quay.io/podified-antelope-centos9/openstack-ceilometer-ipmi@sha256:7f3823163f0b818d3ea72d685dbe3eba4a2a9240ff4bbf687a438d53146270d4 2024-06-27 09:15:16.167813045 +0000 UTC
iscsid quay.io/podified-antelope-centos9/openstack-iscsid@sha256:b20146cd6d00424e4db882767888504dba06ff28f621fa5824693e61277d9244 2024-06-27 09:15:16.279439733 +0000 UTC
logrotate_crond quay.io/podified-antelope-centos9/openstack-cron@sha256:429efae29a08679333f922e006ffe75faa1c75e2bd5431d5ac62bedbae6f00ea 2024-06-27 09:15:16.056319715 +0000 UTC
multipathd quay.io/podified-antelope-centos9/openstack-multipathd@sha256:7a0b89e757c4ad35033ff9900f6db783bb43527242917859acee075805e8373f 2024-06-27 09:15:16.334841798 +0000 UTC
neutron-haproxy-ovnmeta-376b0f04-418a-409a-b9b0-24804b639fb3 quay.io/podified-antelope-centos9/openstack-neutron-metadata-agent-ovn@sha256:fa9fd608d7e26950bdc47a4fa494e8bfe4920b466842ae70efea00e22bbe1482 2024-06-28 08:37:41.926365422 +0000 UTC
node_exporter quay.io/prometheus/node-exporter@sha256:39c642b2b337e38c18e80266fb14383754178202f40103646337722a594d984c 2024-06-27 09:15:17.599357131 +0000 UTC
nova_compute quay.io/podified-antelope-centos9/openstack-nova-compute@sha256:2a284b3c3d7dda709e8c1814a849ff266101e64666754cbbd07928524b66454d 2024-06-27 09:15:16.117323858 +0000 UTC
ovn_controller quay.io/podified-antelope-centos9/openstack-ovn-controller:current-podified 2024-06-27 17:31:43.744346156 +0000 UTC
ovn_metadata_agent quay.io/podified-antelope-centos9/openstack-neutron-metadata-agent-ovn@sha256:fa9fd608d7e26950bdc47a4fa494e8bfe4920b466842ae70efea00e22bbe1482 2024-06-27 09:15:16.488842091 +0000 UTC
```

Reply:
Code bz: https://issues.redhat.com/browse/OSP-32372

Reply:
https://github.com/openstack-k8s-operators/edpm-ansible/pull/688/files merged so I'm going to mark this resolved